### ISSUE TITLE
IConversionOperation.IsChecked is not true from within checked statement

### ISSUE THREAD

**Issue by buyaa-n** [dotnet/roslyn#63951]
`IConversionOperation.IsChecked` returns true as expected from within checked expression, but not returning true when the expression is within checked statement

**Example**:
```cs
class Program
{  
    public unsafe void M1(void* ptr, IntPtr intPtr1)
    {
        checked
        {
            ptr = (void*)intPtr1;  // IConversionOperation.IsChecked is false here - not expected
        }

        ptr = checked((void*)intPtr1); // IConversionOperation.IsChecked is true - as expected
    }
}
```

**Expected Behavior**:
 IConversionOperation.IsChecked should be true when the conversion is within checked statement

**Actual Behavior**:
 IConversionOperation.IsChecked is false

**Comment by jaredpar** [comment]
> IConversionOperation.IsChecked should be true when the conversion is within checked statement

That is not how the property works. It's stating whether the conversion can fail when inside a `checked` block, not whether it occurred in a `checked` block. From the doc comment:

> True if the conversion can fail at runtime with an overflow exception. This corresponds to C# checked and unchecked blocks.

Now this specific code sample will produce `IsChecked == true` on the first instance when compiling against `net7.0`. That is because in `net7.0` we treat `IntPtr` as an alias for `native int` and use it's `checked` operators. Before that though we use the conversion operators on `IntPtr` which are user defined and not classified as `checked` operators.

**Comment by buyaa-n** [comment]
> Now this specific code sample will produce `IsChecked == true` on the first instance when compiling against `net7.0`. That is because in `net7.0` we treat `IntPtr` as an alias for native int and use it's `checked` operators. Before that though we use the conversion operators on `IntPtr` which are user defined and not classified as `checked` operators.

Yes and this is found with the[ analyzer implementation](https://github.com/dotnet/roslyn-analyzers/pull/6153) that specifically checking for that behavioral change and the tests run on `net7.0` and only if the numeric int pointer feature is available. 

That means the code is not producing `IsChecked == true` when compiling against `net7.0`.
